R/shiny.R

```diff
@@ -141,6 +141,8 @@ run <- function(file = "index.Rmd", dir = dirname(file), default_file = NULL,
   # the theme wins out
   if (length(target_file)) {
     format <- output_format_from_yaml_front_matter(read_utf8(target_file))
+    old_theme <- shiny::getCurrentTheme()
+    on.exit(set_current_theme(old_theme), add = TRUE)
     set_current_theme(resolve_theme(format$options$theme))
```

Comment:
Pretty sure when I originally wrote the `set_current_theme()` code I was thinking that since `set_current_theme()` uses `shiny::shinyOptions()`, the previous value would be automatically restored, but it actually doesn't because this call happens outside of an app/session scope (in that case, `shiny::shinyOptions()` is basically like setting `options()`).